homeassistant/components/tankerkoenig/sensor.py

```diff
@@ -0,0 +1,169 @@
+
+DEFAULT_RADIUS = 5
+SCAN_INTERVAL = timedelta(minutes=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Optional(CONF_TYPES, ["e5", "e10", "diesel"]): vol.All(
+            cv.ensure_list, [vol.In(FUEL_TYPES)]
+        ),
+        vol.Inclusive(
+            CONF_LATITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.latitude,
+        vol.Inclusive(
+            CONF_LONGITUDE, "coordinates", "Latitude and longitude must exist together"
+        ): cv.longitude,
+        vol.Optional(CONF_RADIUS, DEFAULT_RADIUS): vol.All(
+            cv.positive_int, vol.Range(min=1)
+        ),
+        vol.Optional(CONF_SCAN_INTERVAL, SCAN_INTERVAL): cv.time_period,
+        vol.Optional(CONF_STATIONS, default=[]): vol.All(cv.ensure_list, [cv.string]),
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set the Tankerkoenig sensor platform up."""
+    fuel_types = config.get(CONF_TYPES, FUEL_TYPES)
+    api_key = config.get(CONF_API_KEY)
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    radius = config.get(CONF_RADIUS, DEFAULT_RADIUS)
+    additional_stations = config.get(CONF_STATIONS, [])
+
+    master = None
+    entities = []
+    _LOGGER.debug("Fetching data for (%s,%s) rad: %s", latitude, longitude, radius)
+    try:
+        data = pytankerkoenig.getNearbyStations(
+            api_key, latitude, longitude, radius, "all", "dist"
+        )
+    except pytankerkoenig.customException as err:
+        data = {"ok": False, "message": err, "exception": True}
+    _LOGGER.debug("Received data: %s", data)
+    if not data["ok"]:
+        _LOGGER.error("Error fetching data from tankerkoenig.de:\n%s", data["message"])
+        _LOGGER.error("Could not setup sensors")
+        return
+
+    if len(data["stations"]) <= 0:
+        _LOGGER.warning("Could not find any station in range")
+    else:
+        for station in data["stations"]:
+            master = add_station(station, api_key, fuel_types, entities, master)
+
+    for additional_station_id in additional_stations:
+        try:
+            additional_station_data = pytankerkoenig.getStationData(
+                api_key, additional_station_id
+            )
+        except pytankerkoenig.customException as err:
+            additional_station_data = {"ok": False, "message": err, "exception": True}
+        if not additional_station_data["ok"]:
+            _LOGGER.warning(
+                "Error when adding station %s: %s",
+                additional_station_id,
+                additional_station_data["message"],
+            )
+        else:
+            master = add_station(
+                additional_station_data["station"],
+                api_key,
+                fuel_types,
+                entities,
+                master,
+            )
```

Comment:
This is already checked in line 68; it is the same api_key; if it worked once, it should also work here.

The loop is going through the list of extra stations requested by the user; printing a warning but ignoring stations with wrong station ID.

Do you mean, if any station is not available (because of typo), the whole setup should fail, instead of showing the stations that could be found?